### PR TITLE
Fix precommit scripts on OSX

### PR DESCRIPTION
**Pull request by franc0is**
JerryScript-DCO-1.0-Signed-off-by: François Baldassari francois@pebble.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
LGTM

File: tools/check-cppcheck.sh [hunk 1]
```diff
@@ -15,7 +15,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-CPPCHECK_JOBS=${CPPCHECK_JOBS:=$(nproc)}
+if [[ "$OSTYPE" == "linux"* ]]; then
+    CPPCHECK_JOBS=${CPPCHECK_JOBS:=$(nproc)}
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    CPPCHECK_JOBS=${CPPCHECK_JOBS:=$(sysctl -n hw.ncpu)}
```
Comment (akiss77) [inline tools/check-cppcheck.sh]: Thanks for getting this work on OSX! This script becomes just a little bit incomplete, however. Please, add a final `else` branch that sets `CPPCHECK_JOBS` to something basic default, just to make sure that the script doesn't break. (E.g., if someone builds on freebsd, or whatever.) I'd go for a default of 1.

**Comment by zherczeg** [comment]
LGTM

**Comment by zherczeg** [comment]
(After the else fix).

**Comment by franc0is** [comment]
Added the else clause and rebased on master.

**Pull request merged**